packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-checkbox-group/gux-form-field-checkbox-group.service.ts

```diff
@@ -0,0 +1,52 @@
+import simulateNativeEvent from '@utils/dom/simulate-native-event';
+
+export function setAllCheckboxInputs(
+  root: HTMLElement,
+  checked: boolean
+): void {
+  getCheckboxInputs(root).forEach(checkboxInput => {
+    if (checkboxInput.checked !== checked && !checkboxInput.disabled) {
+      checkboxInput.checked = checked;
+      simulateNativeEvent(checkboxInput, 'input');
+      simulateNativeEvent(checkboxInput, 'change');
+    }
+  });
+}
+
+export function setParentCheckboxElementCheckedState(
+  root: HTMLElement,
+  mainCheckboxElement: HTMLInputElement
+): void {
+  if (mainCheckboxElement) {
+    const { checkedCheckboxes, totalCheckboxes } = getSelectedColumnCount(root);
+    if (checkedCheckboxes === 0) {
+      mainCheckboxElement.indeterminate = false;
+      mainCheckboxElement.checked = false;
+    } else if (checkedCheckboxes === totalCheckboxes) {
+      mainCheckboxElement.indeterminate = false;
+      mainCheckboxElement.checked = true;
+    } else {
+      mainCheckboxElement.indeterminate = true;
+    }
+  }
+}
+
+export function getSelectedColumnCount(root: HTMLElement): {
```

Comment:
I think this logic still needs to be updated to exclude the parent checkbox I think ? See screenshot below,
![Screenshot 2024-07-17 at 11 15 38](https://github.com/user-attachments/assets/f1521dad-086b-44bc-b3b8-abdc94898797)

I think the parent should be checked also when all the children are checked.